spec/services/proofing/resolution/plugins/aamva_plugin_spec.rb

```diff
@@ -0,0 +1,358 @@
+                failed_result_can_pass_with_additional_verification: true,
+                attributes_requiring_additional_verification: [:address],
+              )
+            end
+
+            it 'makes an AAMVA call' do
+              expect(plugin.proofer).to receive(:proof)
+              call
+            end
+
+            it 'records an SP cost for AAMVA' do
+              expect { call }.to change { sp_cost_count_with_transaction_id }.to(1)
+            end
+          end
+
+          context 'but the failure cannot be covered by AAMVA' do
+            let(:instant_verify_result) do
+              Proofing::Resolution::Result.new(
+                success: false,
+                vendor_name: 'lexisnexis:instant_verify',
+                failed_result_can_pass_with_additional_verification: false,
+              )
+            end
+
+            it 'does not make an AAMVA call' do
+              expect(plugin.proofer).not_to receive(:proof)
+              call
+            end
+
+            it 'does not record an SP cost for AAMVA' do
+              expect { call }.not_to change { sp_cost_count_for_issuer }
+            end
+
+            it 'returns an UnsupportedJurisdiction result' do
+              call.tap do |result|
+                expect(result.success?).to eql(true)
+                expect(result.vendor_name).to eql('UnsupportedJurisdiction')
+              end
+            end
+          end
+        end
+      end
+
+      context 'residential address and id address are different' do
+        let(:applicant_pii) { Idp::Constants::MOCK_IDV_APPLICANT_STATE_ID_ADDRESS }
+
+        context 'InstantVerify succeeded for residential address' do
+          context 'and InstantVerify passed for id address' do
+            let(:instant_verify_result) do
+              Proofing::Resolution::Result.new(
+                success: true,
+                vendor_name: 'lexisnexis:instant_verify',
+              )
+            end
+
+            it 'calls the AAMVA proofer using the state id address' do
+              expect(plugin.proofer).to receive(:proof).with(hash_including(state_id_address))
+              call
+            end
+
+            it 'records an SP cost for AAMVA' do
+              expect { call }.to change { sp_cost_count_with_transaction_id }.to(1)
+            end
+          end
+
+          context 'and InstantVerify failed for state id address' do
+            context 'but the failure can possibly be covered by AAMVA' do
+              let(:instant_verify_result) do
+                Proofing::Resolution::Result.new(
+                  success: false,
+                  vendor_name: 'lexisnexis:instant_verify',
+                  failed_result_can_pass_with_additional_verification: true,
+                  attributes_requiring_additional_verification: [:address],
+                )
+              end
+
+              it 'does not make an AAMVA call because get to yes is not supported' do
```

Comment:
❤️ for this descriptive name -- it makes perfectly clear why we go from "but the failure can possibly be covered by AAMVA" to "does not make an AAMVA call"